repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -4771,20 +4773,83 @@ public void addHasLineage(Set<AtlasEdge> inputOutputEdges, boolean isRestoreEnti
     }
 
 
-    public List<AtlasVertex> linkBusinessPolicy(String policyId, Set<String> linkGuids) {
-        return linkGuids.stream().map(guid -> findByGuid(graph, guid)).filter(Objects::nonNull).filter(ev -> {
-            Set<String> existingValues = ev.getMultiValuedSetProperty(ASSET_POLICY_GUIDS, String.class);
-            return !existingValues.contains(policyId);
-        }).peek(ev -> {
-            Set<String> existingValues = ev.getMultiValuedSetProperty(ASSET_POLICY_GUIDS, String.class);
-            existingValues.add(policyId);
-            ev.setProperty(ASSET_POLICY_GUIDS, policyId);
-            ev.setProperty(ASSET_POLICIES_COUNT, existingValues.size());
+    public AtlasVertex linkBusinessPolicy(final BusinessPolicyRequest.AssetData data) {
+        String assetGuid = data.getAssetId();
+        AtlasVertex vertex = findByGuid(graph, assetGuid);
 
-            updateModificationMetadata(ev);
+        // Retrieve existing policies
+        Set<String> existingCompliant = getVertexPolicies(vertex, ASSET_POLICY_GUIDS);
+        Set<String> existingNonCompliant = getVertexPolicies(vertex, NON_COMPLIANT_ASSET_POLICY_GUIDS);
 
-            cacheDifferentialEntity(ev, existingValues, ev.getMultiValuedSetProperty(NON_COMPLIANT_ASSET_POLICY_GUIDS, String.class));
-        }).collect(Collectors.toList());
+        // Retrieve new policies
+        Set<String> newCompliantRules = data.getComplaintRules();
+        Set<String> newNonCompliantRules = data.getNonComplaintRules();
+        Set<String> newCompliantPolicies = data.getComplaintPolicies();
+
+        // Calculate effective compliant and non-compliant policies
+        Set<String> effectiveCompliant = calculateEffectivePolicies(existingCompliant, newCompliantRules, newCompliantPolicies);
```

Comment:
instead of existingCompliant/newCompliantRules/newCompliantPolicies, can we rewrite as existingCompliantGuids/newCompliantRuleGuids/newCompliantGuids

Overall, suggesting mentioning that all these fields stores policy/Rule GUIDs
This also applies to the fields in the model class BusinessPolicyRequest.AssetData